src/cascadia/WinRTUtils/inc/SafeDispatcherTimer.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+// Par for the course, the XAML timer class is "self-referential". Releasing all references
+// to an instance will not stop the timer. Only calling Stop() explicitly will achieve that.
+struct SafeDispatcherTimer
+{
+    SafeDispatcherTimer() = default;
+    SafeDispatcherTimer(SafeDispatcherTimer const&) = delete;
+    SafeDispatcherTimer& operator=(SafeDispatcherTimer const&) = delete;
+    SafeDispatcherTimer(SafeDispatcherTimer&&) = delete;
+    SafeDispatcherTimer& operator=(SafeDispatcherTimer&&) = delete;
+
+    ~SafeDispatcherTimer()
+    {
+        Destroy();
+    }
+
+    explicit operator bool() const noexcept
+    {
+        return _timer != nullptr;
+    }
+
+    winrt::Windows::Foundation::TimeSpan Interval()
+    {
+        return _getTimer().Interval();
+    }
+
+    void Interval(winrt::Windows::Foundation::TimeSpan const& value)
+    {
+        _getTimer().Interval(value);
+    }
+
+    bool IsEnabled()
+    {
+        return _timer && _timer.IsEnabled();
+    }
+
+    void Tick(winrt::Windows::Foundation::EventHandler<winrt::Windows::Foundation::IInspectable> const& handler)
```

Comment:
Okay so notably, `SafeDispatcherTimer` only supports a single callback at a time for `Tick` (whereas I'm pretty sure that the standard one supports many callbacks) (this is also fine, pretty sure we're not using multiple callbacks on a single timer anyways)

Reply:
My intention was that this avoids accidentally setting the same callback over and over again on the same timer instance. A number of places don't revoke the callbacks after all and just replace the timer instance without calling `Stop()` (i.e. they leak the timer instances).